### PR TITLE
Validation for GH Organizations and GH User Accounts

### PR DESCRIPTION
**Pull request by recena**
[JENKINS-33815](https://issues.jenkins-ci.org/browse/JENKINS-33815)

@reviewbybees, specially @jglick

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java [hunk 1]
```diff
@@ -167,7 +163,11 @@ public String getApiUri() {
             for (GHRepository repo : user.listRepositories()) {
                 add(listener, observer, repo);
             }
+            return;
         }
+
+        listener.getLogger().format("%n%s does not correspond to a known GitHub User Account or Organization%n%n", repoOwner);
+        throw new InterruptedException();
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: :ant: this should be an `AbortException`. `InterruptedException` should be reserved for actual interrupts, which is clearly not the case here, and produces the wrong status message.

Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: And throwing errors without description is bad practise. Just throw AbortException with real description without duplicating it to log before.

Comment (recena) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: @jglick Then, we need a new `Exception` or handling in a different way. As I mentioned [here](https://issues.jenkins-ci.org/browse/JENKINS-33815?focusedCommentId=251922&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-251922) I don't want to show traces to the user.

Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: @recena already working on it. Change it to `AbortException`.

Comment (recena) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: @jglick Is it possible to throw an `Exception` and print its message only in the log file?

Comment (recena) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: @KostyaSha `InterruptedException` does not involve an error following the documentation.

Comment (recena) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: @jglick Done. IMO the `Exception` message should be printed in the log system.

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java [hunk 2]
```diff
@@ -167,7 +164,11 @@ public String getApiUri() {
             for (GHRepository repo : user.listRepositories()) {
                 add(listener, observer, repo);
             }
+            return;
         }
+
+        listener.getLogger().format("%n%s does not correspond to a known GitHub User Account or Organization%n%n", repoOwner);
+        throw new AbortException(repoOwner + "does not correspond to a known GitHub User Account or Organization");
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: :ant: missing space

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java [hunk 3]
```diff
@@ -167,7 +164,11 @@ public String getApiUri() {
             for (GHRepository repo : user.listRepositories()) {
                 add(listener, observer, repo);
             }
+            return;
         }
+
+        listener.getLogger().format("%n%s does not correspond to a known GitHub User Account or Organization%n%n", repoOwner);
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: :ant: redundant, delete. The `AbortException.message` is already displayed in the log.

Comment (recena) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: @jglick I know, I saw your [proposal](https://github.com/jenkinsci/cloudbees-folder-plugin/pull/47/files#diff-64e93ffdb9c069aed757649d11668872L117) after my commit.

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java (hunk 2 above)
Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: space before does...

Comment (recena) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: @KostyaSha @jglick Done.

**Comment by jglick** [comment]
:bee:

**Comment by mslusarczyk** [comment]
:bee:

**Comment by recena** [comment]
@jglick @mslusarczyk Thanks for your time.

**Pull request merged**